src/server/slowlog.cc

```diff
@@ -0,0 +1,72 @@
+// Copyright 2022, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "server/slowlog.h"
+
+namespace dfly {
+
+using namespace std;
+
+constexpr size_t kMaximumSlowlogArgCount = 31;  // 32 - 1 for the command name
+constexpr size_t kMaximumSlowlogArgLength = 128;
+
+SlowLogEntry::SlowLogEntry(uint32_t command_id, uint64_t unix_timestamp,
+                           uint64_t execution_time_micro, size_t original_length,
+                           vector<pair<string, uint32_t>> cmd_args, string client_ip,
+                           string client_name) {
+  command_id_ = command_id;
+  unix_timestamp_ = unix_timestamp;
+  execution_time_micro_ = execution_time_micro;
+  original_length_ = original_length;
+  cmd_args_ = std::move(cmd_args);
+  client_ip_ = std::move(client_ip);
+  client_name_ = std::move(client_name);
+}
+
+size_t SlowLogShard::Length() const {
+  return log_entries_.size();
+}
+
+void SlowLogShard::ChangeLength(const size_t newLength) {
+  log_entries_.set_capacity(newLength);
+}
+
+void SlowLogShard::Reset() {
+  log_entries_.clear();
+}
+
+void SlowLogShard::Add(const string_view command_name, absl::Span<absl::Span<char>> args,
+                       const string_view client_name, const string_view client_ip,
+                       uint64_t execution_time_usec, uint64_t unix_ts) {
+  vector<pair<string, uint32_t>> slowlog_args;
+  size_t slowlog_effective_length = args.size();
+  if (args.size() > kMaximumSlowlogArgCount) {
+    // we store one argument fewer because the last argument is "wasted"
+    // for telling how many further arguments there are
+    slowlog_effective_length = kMaximumSlowlogArgCount - 1;
+  }
+  slowlog_args.reserve(slowlog_effective_length);
+  slowlog_args.emplace_back(command_name, 0);
+
+  for (size_t i = 0; i < slowlog_effective_length; ++i) {
+    string_view arg = facade::ArgS(args, i);
+    size_t extra_bytes = 0;
+    // If any of the arguments is deemed too long, it will be truncated
+    // and the truncated string will be suffixed by the number of truncated bytes in
+    // this format: "... (n more bytes)"
+    size_t extra_bytes_suffix_length = 0;
+    if (arg.size() > kMaximumSlowlogArgLength) {
+      extra_bytes = arg.size() - kMaximumSlowlogArgLength;
```

Comment:
extra_bytes_suffix_length should not be updated as well right?

Reply:
how about keeping the same logic with the original command length and save here the arg.size() and not the extra bytes

Reply:
Hi @UkuLoskit . Please review this comments especially the first one which I think is a bug